### DISCUSSION TITLE
NORM x64 dll build using waf?

### DISCUSSION THREAD

**Discussion by stevek138** [USNavalResearchLaboratory/norm#36] [Q&A] [answered]
Is there a configuration step that needs to be performed to enable the "--msvc_target=x64" option on the waf build? When I run "./waf configure --build-java --msvc_target=x64", as per README-Java.txt, I see an error: "waf: error: no such option: --msvc_target".

I'm specifically trying to build both the .dll and the .so from Ubuntu, for use in an effort integrating NORM via JNI.

Note that (1) the Java build works if I just run "./waf configure --build-java" and (2) all options of the waf build process (as documented in BUILD.TXT) work as expected for the .so.

I've also tried downloading a more current waf (2.0.22) and tried adding the following def configure() in wscript:

        conf.env.MSVC_VERSIONS = ['msvc 10.0', 'msvc 9.0', 'msvc 8.0', 'msvc 7.1', 'msvc 7.0', 'msvc 6.0', 'wsdk 7.0', 'intel 11', 'PocketPC 9.0', 'Smartphone 8.0']
        conf.env.MSVC_TARGETS = ['x64']
        conf.load('msvc')

Both to no avail. I'm new to waf so I may be missing something obvious, and the online documentation references to the --msvc_target flag seem to imply that it should be available with the waf executable download.

Any assistance here would be appreciated.

**Comment by bebopagogo** [comment, marked as answer]
I think that option only works when building on a Windows platform with the Microsoft compiler too chain, etc.  I don’t think waf supports a way to cross-compile a Windows DLL on a Linux platform if that’s what you are trying to do?

**Reply by stevek138** [reply]
Okay, thanks for the quick response. Yes, a cross-compile capability from Ubuntu was what I was looking for. It seems that I misunderstood the capabilities of that flag. I assume that waf runs on Windows via python in a cygwin terminal, so I'll probably just go ahead and configure that.

The quick clarification is very appreciated!
